sentry_streams/src/broadcaster.rs

```diff
@@ -0,0 +1,262 @@
+        traced_with_gil!(|py| {
+            let submitted_messages = Arc::new(Mutex::new(Vec::new()));
+            let submitted_messages_clone = submitted_messages.clone();
+            let submitted_watermarks = Arc::new(Mutex::new(Vec::new()));
+            let submitted_watermarks_clone = submitted_watermarks.clone();
+            let next_step = FakeStrategy::new(
+                submitted_messages.clone(),
+                submitted_watermarks.clone(),
+                true,
+            );
+            let mut step = Broadcaster::new(
+                Box::new(next_step),
+                Route {
+                    source: String::from("source"),
+                    waypoints: vec![],
+                },
+                vec![String::from("branch1"), String::from("branch2")],
+            );
+
+            // Assert MessageRejected adds message to pending
+            let message = Message::new_any_message(
+                build_routed_value(
+                    py,
+                    "test_message".into_py_any(py).unwrap(),
+                    "source",
+                    vec![],
+                ),
+                make_committable(1, 0),
+            );
+            let _ = step.submit(message.clone());
+            assert_eq!(step.pending_messages.len(), 1);
+
+            // Assert MessageRejected adds Watermark to pending
+            let watermark = Message::new_any_message(
+                RoutedValue {
+                    route: Route {
+                        source: String::from("source"),
+                        waypoints: vec![],
+                    },
+                    payload: RoutedValuePayload::WatermarkMessage(WatermarkMessage::Watermark(
+                        Watermark::new(make_committable(2, 0)),
+                    )),
+                },
+                make_committable(2, 0),
+            );
+            let _ = step.submit(watermark);
+            assert_eq!(step.pending_messages.len(), 2);
+
+            // Assert pending message gets sent on repeat submit
+            step.next_step = Box::new(FakeStrategy::new(
+                submitted_messages,
+                submitted_watermarks,
+                false,
+            ));
+            let message = Message::new_any_message(
+                build_routed_value(
+                    py,
+                    "test_message".into_py_any(py).unwrap(),
+                    "source",
+                    vec![],
+                ),
+                make_committable(1, 0),
+            );
+            let _ = step.submit(message.clone());
+            assert_eq!(step.pending_messages.len(), 1);
+            let actual_messages = submitted_messages_clone.lock().unwrap();
+            assert_messages_match(
+                py,
+                vec![
+                    "test_message".into_py_any(py).unwrap(),
+                    "test_message".into_py_any(py).unwrap(),
+                ],
+                actual_messages.deref(),
```

Comment:
This is not enough to test the functionality. It does not check messages are sent on all routes.